weaviate/rbac/permission.go

```diff
@@ -0,0 +1,284 @@
+package rbac
+
+import "github.com/weaviate/weaviate/entities/models"
+
+type Role struct {
+	Permissions
+	Name string
+}
+
+// Permissions holds for all permissions for all resources associated with the role.
+type Permissions struct {
+	Backups     []BackupsPermission
+	Cluster     []ClusterPermission
+	Collections []CollectionsPermission
+	Data        []DataPermission
+	Nodes       []NodesPermission
+	Roles       []RolesPermission
+	Tenants     []TenantsPermission
+	Users       []UsersPermission
+}
+
+// NewRole creates a role with its associated permissions.
+func NewRole(name string, permissionGroups ...PermissionGroup) Role {
+	role := Role{Name: name}
+
+	for _, group := range permissionGroups {
+		group.ExtendRole(&role)
+	}
+	return role
+}
+
+func BackupPermissions(collection string, actions ...string) PermissionGroup {
+	return newPermissionGroup(func(action string) RoleOption {
+		return BackupsPermission{
+			Action:     action,
+			Collection: collection,
+		}
+	}, actions...)
+}
+
+func ClusterPermissions(actions ...string) PermissionGroup {
+	return newPermissionGroup(func(action string) RoleOption {
+		return ClusterPermission{Action: action}
+	}, actions...)
+}
+
+func CollectionsPermissions(collection string, actions ...string) PermissionGroup {
```

Comment:
In general I like this approach, but I don't think this will work for go. The problem being that we will have a tenant filter soonish and then need to change this to
`CollectionsPermissions(collection, tenant string, actions ...string) PermissionGroup` which means a breaking change for go :( Sadly I think we need to have a builder

(I don't understand why they do not have optional arguments, this makes no sense to me)

Reply:
Noted! Will update today.